src/toil/jobStores/awsJobStore.py

```diff
@@ -169,7 +173,9 @@ def delete( self, jobStoreID ):
             log.debug( "Deleting %d file(s) associated with job %s", len( items ), jobStoreID )
             for attempt in retry_sdb( ):
                 with attempt:
-                    self.versions.batch_delete_attributes( { item.name: None for item in items } )
+                    loops = int(math.ceil(float(len(items))/batch_delete_limit))
```

Comment:
Don't use floating-point math for stuff like this. Integer division has the desired truncation properties. Besides, look at 

http://stackoverflow.com/questions/312443/how-do-you-split-a-list-into-evenly-sized-chunks-in-python

for inspiration on how to partition a list. The general idea is to avoid division and multiplication and simple increment an index by the chunk size.